libraries/AP_BattMonitor/AP_BattMonitor.h

```diff
@@ -129,6 +129,7 @@ class AP_BattMonitor
         bool        is_powering_off;           // true when power button commands power off
         bool        powerOffNotified;          // only send powering off notification once
         const struct AP_Param::GroupInfo *var_info;
+        int32_t     time_remaining;            // remaining battery time
```

Comment:
Should we really use signed here?  If so, we really should comment what a negative-time-remaining means.